### PR TITLE
Run3 era for 2023 PbPb UPC re-reco

### PR DESCRIPTION
**Pull request by stahlleiton**
#### PR description:

This PR creates a new era "Run3_2023_UPC" meant for the 2023 PbPb UPC re-reco. It includes the following changes on top of "Run3_2023":
- **High Beta Star (highBetaStar_2018)** modifier: reduces min pt of tracks down to 0.05 GeV.
- **Light by light (egamma_lowPt_exclusive)** modifier: reduces min et of electrons and photons down to 1 GeV
- **run3_upc** modifier:
    - Change pixelPairStepTrackingRegions.originRadius from 0.2 to 0.015 (see [issue](https://github.com/cms-sw/cmssw/issues/43375))
    - Reduce min pt of lowPtGsfElectrons and hiPixelTracks down to 0.05 GeV.
    - Filter on clusters as: # pixel clusters < 10000 and # strip clusters < 30000.
    - Adds: hiPixelTracks, hiCentrality, hiClusterCompatibility, ZDC digis and miniAOD chi2 maps.
    - Remove miniAOD selections for electrons, muons, dEdx, lost tracks and pf candidates.
    - Set miniAOD min pt for slimmed calo jets to 5 GeV.

More details on the validations done are documented in the [slide](https://docs.google.com/presentation/d/1nK3r-i3EwV0YvkpJQb--lPLivD1NCmBTG-XZr__Xr0o/edit?usp=sharing)

@mandrenguyen

### CODE REVIEW

File: Configuration/Eras/python/Era_Run3_UPC_2023_cff.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import FWCore.ParameterSet.Config as cms
+
+from Configuration.Eras.Era_Run3_UPC_cff import Run3_UPC
+from Configuration.Eras.Modifier_run3_egamma_2023_cff import run3_egamma_2023
+
+Run3_UPC_2023 = cms.ModifierChain(Run3_UPC, run3_egamma_2023)
```
Comment (slava77) [inline Configuration/Eras/python/Era_Run3_UPC_2023_cff.py:6]: A large fraction of Eras are named `RunX_YYYY_mod` is it practical to create a bubble in the implementation of the pp era (here you re-implement the `Run3` -> `Run3_2023` detail).  ... why do we need `Run3_UPC`?

File: RecoEgamma/EgammaElectronProducers/python/lowPtGsfElectronSeeds_cfi.py [hunk 2]
```diff
@@ -52,3 +52,6 @@ def thresholds( wp ) :
 
 from Configuration.ProcessModifiers.pp_on_AA_cff import pp_on_AA
 pp_on_AA.toModify(lowPtGsfElectronSeeds,MinPtThreshold = 5.0)
+
+from Configuration.Eras.Modifier_run3_upc_cff import run3_upc
+run3_upc.toModify(lowPtGsfElectronSeeds, ModelThresholds = thresholds("VL"), MinPtThreshold = 0.05)
```
Comment (slava77) [inline RecoEgamma/EgammaElectronProducers/python/lowPtGsfElectronSeeds_cfi.py:57]: shouldn't everything egamma just follow `egamma_lowPt_exclusive` modifier ?

File: RecoTracker/IterativeTracking/python/PixelPairStep_cff.py [hunk 3]
```diff
@@ -92,6 +92,8 @@
      originRadius = 0.2,
      fixedError   = 4.
 ))
+from Configuration.Eras.Modifier_highBetaStar_2023_cff import highBetaStar_2023
+highBetaStar_2023.toModify(pixelPairStepTrackingRegions,RegionPSet = dict(originRadius = 0.015))
```
Comment (slava77) [inline RecoTracker/IterativeTracking/python/PixelPairStep_cff.py:96]: I'd add a comment about the issue in the efficiency recovery; once that problem is addressed I expect that `highBetaStar_2018` and `highBetaStar_2023` will be identical.
It may even be more practical to use `(highBetaStar_2018 & run3_upc)` here.

IIRC, the only highBeta run in 2023 was after the BPIX problem appeared, but as I recall those fills had no pixel tracker readout. @sroychow @mmusich please check and correct me if I'm wrong

**Comment by slava77** [comment]
it would be nice to have a test workflow

File: Configuration/Eras/python/Era_Run3_UPC_2023_cff.py (hunk 1 above)
Comment (stahlleiton) [reply Configuration/Eras/python/Era_Run3_UPC_2023_cff.py:6]: Not needed at the moment, so I will remove Run3_UPC

**Comment by stahlleiton** [comment]
> it would be nice to have a test workflow

Is there instructions on how to set up a test workflow? (never done it before)

File: RecoTracker/IterativeTracking/python/PixelPairStep_cff.py (hunk 3 above)
Comment (stahlleiton) [reply RecoTracker/IterativeTracking/python/PixelPairStep_cff.py]: Changed to the approach suggested using "(highBetaStar_2018 & run3_upc)"

File: RecoEgamma/EgammaElectronProducers/python/lowPtGsfElectronSeeds_cfi.py (hunk 2 above)
Comment (stahlleiton) [reply RecoEgamma/EgammaElectronProducers/python/lowPtGsfElectronSeeds_cfi.py]: Moved all the egamma related changes to egamma_lowPt_exclusive

**Comment by slava77** [comment]
> > it would be nice to have a test workflow
> 
> Are there instructions on how to set up a test workflow? (never done it before)

for HI workflows it seems like the practice was to go in 
 in Configuration/PyReleaseValidation/python: 
 in a combination of relval_steps.py (defines the steps) with relval_standard.py used to define the actual numbered workflow from the steps.
I suggest to look for Run3 and AA in the steps file to see the expected pattern; hopefully it's clear enough.

[ pp standard MC uses upgradeWorkflowComponents.py ]

**Comment by mandrenguyen** [comment]
I suppose you'll want a real data workflow.  In that case you can use the following example:
https://github.com/cms-sw/cmssw/pull/40341
If we want MC instead, something like this might be more useful:
https://github.com/cms-sw/cmssw/pull/41095

File: RecoTracker/IterativeTracking/python/PixelPairStep_cff.py (hunk 3 above)
Comment (mmusich) [reply RecoTracker/IterativeTracking/python/PixelPairStep_cff.py]: > IIRC, the only highBeta run in 2023 was after the BPIX problem appeared, but as I recall those fills had no pixel tracker readout. @sroychow @mmusich please check and correct me if I'm wrong

based on [OMS](https://cmsoms.cern.ch/cms/fills/summary?cms_fill_to=9322&cms_fill_from=9000&cms_fill_stableOnly=true) this year we had the fills in the interval 9095-9168 with non-nominal beta* (120cm). All of them happened in era >= Run2023E (which is after the pixel readout incident).
As far as I can tell in none of these fills either Pixel or Strip tracker was in global DAQ excepted fills 9095 and 9128. 
But in any case I don't recall the Tier-0 being moved to the highBetaStar scenario.

Comment (mmusich) [reply RecoTracker/IterativeTracking/python/PixelPairStep_cff.py]: > once that problem is addressed I expect that highBetaStar_2018 and highBetaStar_2023 will be identical.

Incidentally (and certainly for outside of this particular PR), what's so intrinsically about 2018 of the `highBetaStar_2018` modifier? Can't it just be `highBetaStar` (or `highBetaStar_Phase1` to signal it applies to phase-1 tracking)?

Comment (slava77) [reply RecoTracker/IterativeTracking/python/PixelPairStep_cff.py]: `highBetaStar_Phase1` makes sense

2018 is likely the first time it made it to the configs

File: Configuration/Eras/python/Era_Run3_UPC_2023_cff.py [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import FWCore.ParameterSet.Config as cms
+
+from Configuration.Eras.Era_Run3_cff import Run3
+from Configuration.ProcessModifiers.egamma_lowPt_exclusive_cff import egamma_lowPt_exclusive
+from Configuration.Eras.Modifier_highBetaStar_2018_cff import highBetaStar_2018
+from Configuration.Eras.Modifier_run3_upc_cff import run3_upc
+from Configuration.Eras.Modifier_run3_egamma_2023_cff import run3_egamma_2023
+
+Run3_UPC_2023 = cms.ModifierChain(Run3, egamma_lowPt_exclusive, highBetaStar_2018, run3_upc, run3_egamma_2023)
```
Comment (makortel) [inline Configuration/Eras/python/Era_Run3_UPC_2023_cff.py:9]: `Run3_2023_UPC` would follow better the existing naming convention. How about using `Run3_2023` as the starting point?

Comment (stahlleiton) [reply Configuration/Eras/python/Era_Run3_UPC_2023_cff.py]: I implemented your suggestion

**Comment by stahlleiton** [comment]
Implemented two new data workflows:
141.901: Run3_2023_UPC reconstruction on HI Forward dataset
141.902: Run3_2023 (pp) reconstruction on HI Forward dataset

Both can be used to compare against each other

**Comment by mandrenguyen** [comment]
please test workflow 141.901, 141.902

**Comment by stahlleiton** [comment]
Not sure I understand what failed or how to see the comparison plots between the two workflows.

File: Configuration/StandardSequences/python/Eras.py [hunk 5]
```diff
@@ -83,10 +84,10 @@ def __init__(self):
                            'trackingPhase2PU140','highBetaStar_2018',
                            'tracker_apv_vfp30_2016', 'pf_badHcalMitigationOff',
                            'run2_miniAOD_80XLegacy','run2_miniAOD_94XFall17',
-                           'run3_miniAOD_12X',
```
Comment (mmusich) [inline Configuration/StandardSequences/python/Eras.py:86]: why is this removed?

**Comment by mmusich** [comment]
> Not sure I understand what failed or how to see the comparison plots between the two workflows.

@stahlleiton you (accidentally?) removed one line in the `Eras.py` file, see https://github.com/cms-sw/cmssw/pull/43378#discussion_r1409202154

**Comment by mandrenguyen** [comment]
@stahlleiton It might be a transient failure.  We can relaunch the tests after you respond to Marco's point.

**Comment by stahlleiton** [comment]
> > Not sure I understand what failed or how to see the comparison plots between the two workflows.
> 
> @stahlleiton you (accidentally?) removed one line in the `Eras.py` file, see https://github.com/cms-sw/cmssw/pull/43378#discussion_r1409202154

Hmm.., seems I indeed accidentally deleted that line (probably while fixing the merge conflicts). I will move to the latest CMSSW dev release and rebase my commits, while adding back this line.

**Comment by mmusich** [comment]
@mandrenguyen 

> It might be a transient failure. 

it's most definitely not. You can convince yourself just opening the [log file](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3387b6/36144/runTheMatrixINPUT-results/140.201_RunJetMET2022D_reMINI/step2_RunJetMET2022D_reMINI.log)

```
PAT,ENDJOB
Traceback (most recent call last):
  File "/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/43378/36144/CMSSW_14_0_X_2023-11-28-2300/bin/el8_amd64_gcc12/cmsDriver.py", line 40, in <module>
    run()
  File "/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/43378/36144/CMSSW_14_0_X_2023-11-28-2300/bin/el8_amd64_gcc12/cmsDriver.py", line 11, in run
    options = OptionsFromCommandLine()
  File "/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/43378/36144/CMSSW_14_0_X_2023-11-28-2300/src/Configuration/Applications/python/cmsDriverOptions.py", line 36, in OptionsFromCommandLine
    options=OptionsFromItems(sys.argv[1:])
  File "/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/43378/36144/CMSSW_14_0_X_2023-11-28-2300/src/Configuration/Applications/python/cmsDriverOptions.py", line 267, in OptionsFromItems
    raise Exception( "'%s' is not a valid option for '--era'. Valid options are %s." % (eraName, validOptions) )
Exception: 'run3_miniAOD_12X' is not a valid option for '--era'. Valid options are 'Run1_pA', 'Run1_peripheralPbPb', 'Run2_50ns', 'Run2_50ns_HIPM', 'Run2_25ns', 'Run2_25ns_HIPM', 'Run2_25ns_peripheralPbPb', 'Run2_HI', 'Run2_2016', 'Run2_2016_HIPM', 'Run2_2016_trackingLowPU', 'Run2_2016_pA', 'Run2_2017', 'Run2_2017_noMkFit', 'Run2_2017_FastSim', 'Run2_2017_trackingRun2', 'Run2_2017_trackingLowPU', 'Run2_2017_pp_on_XeXe', 'Run2_2017_ppRef', 'Run2_2018', 'Run2_2018_FastSim', 'Run2_2018_pp_on_AA', 'Run2_2018_pp_on_AA_noHCALmitigation', 'Run2_2018_highBetaStar', 'Run2_2018_noMkFit', 'Run3', 'Run3_2023', 'Run3_noMkFit', 'Run3_pp_on_PbPb', 'Run3_pp_on_PbPb_approxSiStripClusters', 'Run3_pp_on_PbPb_2023', 'Run3_pp_on_PbPb_approxSiStripClusters_2023', 'Run3_dd4hep', 'Run3_DDD', 'Run3_FastSim', 'Run3_2023_FastSim', 'Run3_2023_ZDC', 'Run3_2023_UPC', 'Phase2', 'Phase2C9', 'Phase2C10', 'Phase2C11', 'Phase2C12', 'Phase2C9_dd4hep', 'Phase2C10_dd4hep', 'Phase2C11_dd4hep', 'Phase2C11I13', 'Phase2C12_dd4hep', 'Phase2C11M9', 'Phase2C11I13M9', 'Phase2C11I13T25M9', 'Phase2C11I13T26M9', 'Phase2C17I13M9', 'Phase2C20I13M9', 'run2_common', 'run2_25ns_specific', 'run2_50ns_specific', 'run2_HI_specific', 'stage1L1Trigger', 'fastSim', 'peripheralPbPb', 'pA_2016', 'run2_HE_2017', 'stage2L1Trigger', 'stage2L1Trigger_2017', 'stage2L1Trigger_2018', 'stage2L1Trigger_2021', 'run2_HF_2017', 'run2_HCAL_2017', 'run2_HEPlan1_2017', 'run2_HB_2018', 'run2_HE_2018', 'run3_HB', 'run3_HFSL', 'run3_common', 'run3_RPC', 'phase1Pixel', 'run3_GEM', 'run2_GEM_2017', 'run2_CSC_2018', 'phase2_common', 'phase2_tracker', 'phase2_muon', 'phase2_GEM', 'phase2_GE0', 'phase2_hgcal', 'phase2_timing', 'phase2_hfnose', 'phase2_hgcalV10', 'phase2_hgcalV11', 'phase2_hgcalV12', 'phase2_timing_layer', 'phase2_etlV4', 'phase2_hcal', 'phase2_ecal', 'phase2_ecal_devel', 'phase2_trigger', 'phase2_squarePixels', 'phase2_3DPixels', 'trackingLowPU', 'trackingPhase1', 'ctpps', 'ctpps_2016', 'ctpps_2017', 'ctpps_2018', 'ctpps_2022', 'trackingPhase2PU140', 'highBetaStar_2018', 'tracker_apv_vfp30_2016', 'pf_badHcalMitigationOff', 'run2_miniAOD_80XLegacy', 'run2_miniAOD_94XFall17', 'run2_nanoAOD_106Xv2', 'run3_nanoAOD_122', 'run3_nanoAOD_124', 'run3_ecal_devel', 'run3_upc', 'hcalHardcodeConditions', 'hcalSkipPacker', 'run2_HLTconditions_2016', 'run2_HLTconditions_2017', 'run2_HLTconditions_2018', 'bParking', 'run2_2017_noTrackingModifier', 'trackingMkFitProd'.
```

this is what causing the integration tests to fail...

**Comment by mandrenguyen** [comment]
ok, conclusion remains the same.

> @mandrenguyen
> 
> > It might be a transient failure.
> 
> it's most definitely not. You can convince yourself just opening the [log file](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3387b6/36144/runTheMatrixINPUT-results/140.201_RunJetMET2022D_reMINI/step2_RunJetMET2022D_reMINI.log)
> 
> ```
> PAT,ENDJOB
> Traceback (most recent call last):
>   File "/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/43378/36144/CMSSW_14_0_X_2023-11-28-2300/bin/el8_amd64_gcc12/cmsDriver.py", line 40, in <module>
>     run()
>   File "/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/43378/36144/CMSSW_14_0_X_2023-11-28-2300/bin/el8_amd64_gcc12/cmsDriver.py", line 11, in run
>     options = OptionsFromCommandLine()
>   File "/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/43378/36144/CMSSW_14_0_X_2023-11-28-2300/src/Configuration/Applications/python/cmsDriverOptions.py", line 36, in OptionsFromCommandLine
>     options=OptionsFromItems(sys.argv[1:])
>   File "/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/43378/36144/CMSSW_14_0_X_2023-11-28-2300/src/Configuration/Applications/python/cmsDriverOptions.py", line 267, in OptionsFromItems
>     raise Exception( "'%s' is not a valid option for '--era'. Valid options are %s." % (eraName, validOptions) )
> Exception: 'run3_miniAOD_12X' is not a valid option for '--era'. Valid options are 'Run1_pA', 'Run1_peripheralPbPb', 'Run2_50ns', 'Run2_50ns_HIPM', 'Run2_25ns', 'Run2_25ns_HIPM', 'Run2_25ns_peripheralPbPb', 'Run2_HI', 'Run2_2016', 'Run2_2016_HIPM', 'Run2_2016_trackingLowPU', 'Run2_2016_pA', 'Run2_2017', 'Run2_2017_noMkFit', 'Run2_2017_FastSim', 'Run2_2017_trackingRun2', 'Run2_2017_trackingLowPU', 'Run2_2017_pp_on_XeXe', 'Run2_2017_ppRef', 'Run2_2018', 'Run2_2018_FastSim', 'Run2_2018_pp_on_AA', 'Run2_2018_pp_on_AA_noHCALmitigation', 'Run2_2018_highBetaStar', 'Run2_2018_noMkFit', 'Run3', 'Run3_2023', 'Run3_noMkFit', 'Run3_pp_on_PbPb', 'Run3_pp_on_PbPb_approxSiStripClusters', 'Run3_pp_on_PbPb_2023', 'Run3_pp_on_PbPb_approxSiStripClusters_2023', 'Run3_dd4hep', 'Run3_DDD', 'Run3_FastSim', 'Run3_2023_FastSim', 'Run3_2023_ZDC', 'Run3_2023_UPC', 'Phase2', 'Phase2C9', 'Phase2C10', 'Phase2C11', 'Phase2C12', 'Phase2C9_dd4hep', 'Phase2C10_dd4hep', 'Phase2C11_dd4hep', 'Phase2C11I13', 'Phase2C12_dd4hep', 'Phase2C11M9', 'Phase2C11I13M9', 'Phase2C11I13T25M9', 'Phase2C11I13T26M9', 'Phase2C17I13M9', 'Phase2C20I13M9', 'run2_common', 'run2_25ns_specific', 'run2_50ns_specific', 'run2_HI_specific', 'stage1L1Trigger', 'fastSim', 'peripheralPbPb', 'pA_2016', 'run2_HE_2017', 'stage2L1Trigger', 'stage2L1Trigger_2017', 'stage2L1Trigger_2018', 'stage2L1Trigger_2021', 'run2_HF_2017', 'run2_HCAL_2017', 'run2_HEPlan1_2017', 'run2_HB_2018', 'run2_HE_2018', 'run3_HB', 'run3_HFSL', 'run3_common', 'run3_RPC', 'phase1Pixel', 'run3_GEM', 'run2_GEM_2017', 'run2_CSC_2018', 'phase2_common', 'phase2_tracker', 'phase2_muon', 'phase2_GEM', 'phase2_GE0', 'phase2_hgcal', 'phase2_timing', 'phase2_hfnose', 'phase2_hgcalV10', 'phase2_hgcalV11', 'phase2_hgcalV12', 'phase2_timing_layer', 'phase2_etlV4', 'phase2_hcal', 'phase2_ecal', 'phase2_ecal_devel', 'phase2_trigger', 'phase2_squarePixels', 'phase2_3DPixels', 'trackingLowPU', 'trackingPhase1', 'ctpps', 'ctpps_2016', 'ctpps_2017', 'ctpps_2018', 'ctpps_2022', 'trackingPhase2PU140', 'highBetaStar_2018', 'tracker_apv_vfp30_2016', 'pf_badHcalMitigationOff', 'run2_miniAOD_80XLegacy', 'run2_miniAOD_94XFall17', 'run2_nanoAOD_106Xv2', 'run3_nanoAOD_122', 'run3_nanoAOD_124', 'run3_ecal_devel', 'run3_upc', 'hcalHardcodeConditions', 'hcalSkipPacker', 'run2_HLTconditions_2016', 'run2_HLTconditions_2017', 'run2_HLTconditions_2018', 'bParking', 'run2_2017_noTrackingModifier', 'trackingMkFitProd'.
> ```
> 
> this is what causing the integration tests to fail...

**Comment by stahlleiton** [comment]
I just rebased to latest release (CMSSW_14_0_X_2023-11-29-1100) to avoid further merge conflicts and added back the line in Eras.py

**Comment by mmusich** [comment]
test parameters:
   * workflows = 141.901, 141.902

**Comment by mmusich** [comment]
please test

**Comment by mandrenguyen** [comment]
This PR looks ok to me.
@cms-sw/tracking-pog-l2 I lost track, were you convinced that the changes to the pixel pair tracking region make sense and are the right way to deal with the observed inefficiency?
Or is there still something you would like @stahlleiton to check?

**Comment by slava77** [comment]
> This PR looks ok to me. @cms-sw/tracking-pog-l2 I lost track, were you convinced that the changes to the pixel pair tracking region make sense and are the right way to deal with the observed inefficiency? Or is there still something you would like @stahlleiton to check?

If you are referring to #43375 , I'm not aware of updates on that side.

IIUC, the high `Beta*` fills have a larger beam spot profile. Looking at run 373061 with 19.2 m (instead of 1.2 m) the x/y width is around 60 µm (vs nominal of around 8-10 µm), which I suppose is radius around 80 µm.
In your setup you are switching to 150 µm cut on the origin radius, which is likely going to lead to some inefficiency even for prompt tracks if it were for the actual high beta*.

Apparently the HI BS width in 2023 is just above 10 µm. So, the update in this PR is still OK for your case promptish tracks.

Coupled with UPC (`highBetaStar_2018 & run3_upc`), I think this PR update is OK.

**Comment by mandrenguyen** [comment]
+1

**Comment by stahlleiton** [comment]
Are there pending comments from pdmv, xpog and upgrade, or is the PR fine to be approved?

**Comment by mmusich** [comment]
> Would it be worth of opening an issue about these?

here it is: https://github.com/cms-sw/cmssw/issues/43488

**Comment by stahlleiton** [comment]
Thanks for the feedback.
@cms-sw/pdmv-l2 and @cms-sw/xpog-l2: Is the PR fine to be approved?

**Comment by stahlleiton** [comment]
@sunilUIET , @AdrianoDee , @simonepigazzini , @vlimant : Can you please sign this PR?

**Comment by AdrianoDee** [comment]
+pdmv

**Comment by vlimant** [comment]
+1

**Comment by antoniovilela** [comment]
+1
